### PR TITLE
ADM-729[backend]feat: can get all history for card

### PR DESCRIPTION
**Pull request by BoBoDai**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 1]
```diff
@@ -530,10 +529,22 @@ private List<String> getAssigneeSet(StoryPointsAndCycleTimeRequest request, URI
 	}
 
 	private List<String> getHistoricalAssignees(URI baseUrl, String cardKey, String token) {
-		CardHistoryResponseDTO jiraCardHistory = jiraFeignClient.getJiraCardHistory(baseUrl, cardKey, token);
+		CardHistoryResponseDTO jiraCardHistory = getCardAllHistory(baseUrl, cardKey, 0, token);
 		return jiraCardHistory.getItems().stream().map(item -> item.getActor().getDisplayName()).distinct().toList();
 	}
 
+	private CardHistoryResponseDTO getCardAllHistory(URI baseUrl, String cardKey, int startAt, String token) {
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:536]: `getCardAllHistory` or `getCardHistory`?
what is all history

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java [hunk 2]
```diff
@@ -530,10 +529,22 @@ private List<String> getAssigneeSet(StoryPointsAndCycleTimeRequest request, URI
...
+		int pageCount = 100;
```
Comment (guzhongren) [inline backend/src/main/java/heartbeat/service/board/jira/JiraService.java:537]: how to handle the page of `200`.....

File: backend/src/test/java/heartbeat/service/jira/JiraServiceTest.java [hunk 3]
```diff
@@ -349,8 +349,37 @@ void shouldCallJiraFeignClientAndThrowNotFoundExceptionWhenGetJiraBoardConfig()
...
+			.thenReturn(new CardHistoryResponseDTO(true, Collections.emptyList()));
+		when(jiraFeignClient.getTargetField(baseUrl, "project key", token))
+			.thenReturn(FIELD_RESPONSE_BUILDER().build());
+
+		assertThatCode(() -> jiraService.getJiraConfiguration(boardTypeJira, boardRequestParam))
+			.doesNotThrowAnyException();
+	}
+
+	@Test
+	void shouldCallJiraFeignClientTwiceWhenHistoryDataToManyAndThrowNotFoundExceptionWhenGetJiraBoardConfig()
```
Comment (guzhongren) [inline backend/src/test/java/heartbeat/service/jira/JiraServiceTest.java:362]: 2 when?

**Review by guzhongren — CHANGES_REQUESTED** [review]
Change with the comments

File: backend/src/main/java/heartbeat/service/board/jira/JiraService.java (hunk 2 above)
Comment (BoBoDai) [reply backend/src/main/java/heartbeat/service/board/jira/JiraService.java:537]: Get all data recursively, 100 items each time

**Review by weiraneve — APPROVED** [review]

**Review by guzhongren — APPROVED** [review]

**Pull request merged**